batch/batch/cloud/azure/driver/create_instance.py

```diff
@@ -202,6 +203,8 @@ def create_vm_config(
 # Forbid outgoing requests to cluster-internal IP addresses
 INTERNET_INTERFACE=eth0
 iptables --append FORWARD --out-interface $INTERNET_INTERFACE ! --destination 10.128.0.0/16 --jump ACCEPT
+# Allow incoming packets to job network namespaces only from the internet
+iptables --append FORWARD --in-interface $INTERNET_INTERFACE --destination 172.20.0.0/15 --jump ACCEPT
```

Comment:
Mmmm. I'm a bit confused. How does the internet interface receive a packet destined for 172.20.0.1? The internet does not know this machine can serve that IP address.

Reply:
OK, I think I get it. This page helped: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/4/html/security_guide/s1-firewall-ipt-fwd

The PREROUTING chain effectively rewrites the IP address *and then* puts the packet through the FORWARD chain. That's how we the public internet interface generates an internal IP address. In fact, since Google's routing system would never deliver us packets for 172.20.0.1, we *know* that this packet came from our own PREROUTING rules. Does that all sound right to you?